app/src/main/java/org/wikipedia/edit/richtext/SyntaxHighlighter.kt

```diff
@@ -0,0 +1,250 @@
+                                newSpanInfo = spanStack.pop()
+                                newSpanInfo.end = i + syntaxItem.startSymbol.length
+                                spansToSet.add(newSpanInfo)
+                            } else {
+                                val sp = syntaxItem.spanStyle.createSpan(context, i, syntaxItem)
+                                spanStack.push(sp)
+                            }
+                            i += syntaxItem.startSymbol.length
+                            incrementDone = true
+                        }
+                    } else {
+                        var pass = true
+                        for (j in syntaxItem.startSymbol.indices) {
+                            if (text[i + j] != syntaxItem.startSymbol[j]) {
+                                pass = false
+                                break
+                            }
+                        }
+                        if (pass) {
+                            val sp = syntaxItem.spanStyle.createSpan(context, i, syntaxItem)
+                            spanStack.push(sp)
+                            i += syntaxItem.startSymbol.length
+                            incrementDone = true
+                        }
+                        // skip the check of end symbol when start symbol is found at end of the text
+                        if (i + syntaxItem.startSymbol.length > text.length) {
+                            continue
+                        }
+                        pass = true
+                        for (j in syntaxItem.endSymbol.indices) {
+                            if (text[i + j] != syntaxItem.endSymbol[j]) {
+                                pass = false
+                                break
+                            }
+                        }
+                        if (pass) {
+                            if (spanStack.size > 0 && spanStack.peek().syntaxRule == syntaxItem) {
+                                newSpanInfo = spanStack.pop()
+                                newSpanInfo.end = i + syntaxItem.endSymbol.length
+                                spansToSet.add(newSpanInfo)
+                            }
+                            i += syntaxItem.endSymbol.length
+                            incrementDone = true
+                        }
+                    }
+                }
+                if (cancelled) {
+                    break
+                }
+                if (!incrementDone) {
+                    i++
+                }
+            }
+            return spansToSet
+        }
+    }
+
+    private inner class SyntaxHighlightSearchMatchesTask constructor(text: Editable, searchText: String?, private val selectedMatchResultPosition: Int) : Callable<List<SpanExtents>> {
+        private val searchText: String = StringUtils.lowerCase(searchText).orEmpty()
+        private val text: String = StringUtils.lowerCase(text.toString()).orEmpty()
```

Comment:
Would it be possible to replace these to `.orEmpty().toLowerCase(Locale.ROOT)`?

Reply:
damnit!